fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -2612,7 +2612,7 @@ public int reindex(FHIRPersistenceContext context, OperationOutcome.Builder oper
             // It's possible this is a deadlock exception, in which case it could be considered retryable
             if (dax.isTransactionRetryable()) {
                 log.log(Level.WARNING, "Retryable error while performing reindex" + (rir != null ? (" of FHIR Resource '" + rir.getResourceType() + "/" + rir.getLogicalId() + "'") : ""), dax);
-                FHIRPersistenceDataAccessException fpx = new FHIRPersistenceDataAccessException("Data access error while performing a reindex operation.");
```

Comment:
I think that we were purposefully not including the original exception here because we were worried it could contain sensitive data in the SQL, so we were logging it here and then throwing a more generic error.  I think that was a change that @punktilious made a while back and I'm not certain its needed here, so I'll defer to y'all.

Reply:
Yeah, I'm thinking we don't need the causedBy here because its part of the WARNING log statement in the previously line

Reply:
I need the original exception.

Reply:
Oh, do we walk through the causedBy exceptions somewhere?  I didn't see that.

Reply:
Yes, in the FHIRRestHelper I check to see if the causedby is LockException and is a deadlock in order to execute a backoff.